overlord/snapstate/snapmgr.go

```diff
@@ -309,27 +361,33 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	meter := &TaskProgressAdapter{task: t}
 
-	var auther store.Authenticator
-	if ss.UserID > 0 {
-		st.Lock()
-		user, err := auth.User(st, ss.UserID)
-		st.Unlock()
+	st.Lock()
+	auther, err := autherForUserID(st, ss.UserID)
+	st.Unlock()
+	if err != nil {
+		return err
+	}
+
+	var storeInfo *snap.Info
+	if ss.UpdateInfo != nil {
+		storeInfo = &snap.Info{
```

Comment:
my suggesting about DownloadInfo (embedded in Info) avoids having to create this half info here, also allows only snap and store to need to care what we need for a download